### ISSUE TITLE
Accessibility issue on TableToolBar search

### ISSUE THREAD

**Issue by palak1459** [carbon-design-system/carbon#8260]
## Environment

> Operating system

Mac

> Browser

Chrome, Firefox

> Assistive technology used to verify

IBM Equal Access Accessibility Checker

## Detailed description

> What version of the Carbon Design System are you using?

"@carbon/ibm-security": "1.31.0",
"carbon-components": "10.7.4",
"@types/carbon-components-react": "7.6.5",

> What did you expect to happen?

No accessibility issue on TableToolBar search

> What happened instead?

Found one accessibility violation.

> What WCAG 2.1 checkpoint does the issue violate?

_The value "1" of the 'for' attribute is not the 'id' of a valid <input> element_

## Additional information

- Screenshots or code
<img width="1792" alt="Screenshot 2021-04-01 at 9 32 31 AM" src="https://user-images.githubusercontent.com/71092117/113241870-316b5200-92cd-11eb-86d6-b1c8c8f1f14b.png">
<img width="386" alt="Screenshot 2021-04-01 at 9 33 24 AM" src="https://user-images.githubusercontent.com/71092117/113241926-51027a80-92cd-11eb-9319-0d67f08acc94.png">

**Comment by tw15egan** [comment]
Do you have a link to a reproducible example on [CodeSandbox](https://codesandbox.io/s/dark-worker-szrpt)? I'm unable to reproduce the a11y violation in our storybook

**Comment by joshblack** [comment]
Going to close this one out as needs more info. If you could share steps to reproduce the problem we can take a look at this!
